File: rb/features.rb
# AddressLookupService SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module AddressLookupServiceFeatures
  def self.make_feature(name)
    case name
    when "base"
      AddressLookupServiceBaseFeature.new
    when "test"
      AddressLookupServiceTestFeature.new
    else
      AddressLookupServiceBaseFeature.new
    end
  end
end

File: py/features.py
# AddressLookupService SDK feature factory

from feature.base_feature import AddressLookupServiceBaseFeature
from feature.test_feature import AddressLookupServiceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AddressLookupServiceBaseFeature(),
        "test": lambda: AddressLookupServiceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
